Packs/Bonusly/Integrations/Bonusly/Bonusly.py

```diff
@@ -0,0 +1,350 @@
+    result = resp.get('result')
+    readable_output = tableToMarkdown('Latest Updates From Bonus.ly', result)
+    outputs = {
+        'Bonusly.Bonus': result
+    }
+
+    return (
+        readable_output,
+        outputs,
+        resp  # raw response - the original response
+    )
+
+
+def bonusly_update_bonus_command(client, args):
+    resp = client.bonusly_update_bonus_http_request(args)
+    result = resp.get('result')
+    readable_output = tableToMarkdown('Message Deleted From Bonus.ly', result)
+    outputs = {
+        'Bonusly.Bonus': result
+    }
+
+    return (
+        readable_output,
+        outputs,
+        resp  # raw response - the original response
+    )
+
+
+def bonusly_delete_bonus_command(client, args):
+    resp = client.bonusly_delete_bonus_http_request(args)
+    result_text = resp.get('message')
+    res_message = []
+    res_message_table = {'Message': result_text}
+    res_message.append(res_message_table)
+    readable_output = tableToMarkdown('Latest Updates From Bonus.ly', res_message_table)
+    outputs = {
+        'Bonusly.Bonus.message': result_text
+    }
+
+    return (
+        readable_output,
+        outputs,
+        resp  # raw response - the original response
+    )
+
+
+def fetch_incidents(client, last_run, first_fetch_time, fetch_params):
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client (Client): client
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+        first_fetch_time (dateparser.time): If last_run is None then fetch all incidents since first_fetch_time
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+        :param fetch_params params from integration settings
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first time fetch
+    if last_fetch is None:
+        last_fetch = dateparser.parse(first_fetch_time)
+    else:
+        last_fetch = dateparser.parse(last_fetch)
+
+    latest_created_time = last_fetch
+    incidents = []
+    last_fetch_date = last_fetch.strftime('%Y-%m-%dT%H:%M:%SZ')
+    items = client.list_incidents(last_fetch_date=last_fetch_date, fetch_params=fetch_params)
+    for item in items:
+        incident_created_time = dateparser.parse(item['created_at'])
+        incident = {
+            'name': 'New Bonusly - ' + item['reason'],
+            'occurred': incident_created_time.strftime(DATE_FORMAT),
+            'type': fetch_params.get('incidentType'),
+            'rawJSON': json.dumps(item)
+        }
+
+        # Update last run and add incident if the incident is newer than last fetch
+        if incident_created_time.timestamp() > latest_created_time.timestamp():
+            latest_created_time = incident_created_time
+            incidents.append(incident)
```

Comment:
```suggestion
if incident_created_time.timestamp() > current_fetch_time.timestamp():
            incidents.append(incident)
```
also create a variable called `current_fetch_time = latest_created_time` in line 269.

together this will make sure only new incidents are entered even if the bonuses are fetched out of order.